cli/containerizedengine/hostpaths.go

```diff
@@ -0,0 +1,133 @@
+package containerizedengine
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"path"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	"github.com/docker/cli/pkg/containerized"
+	"github.com/docker/docker/pkg/stringid"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func (c baseClient) verifyDockerConfig(ctx context.Context, configFile string, image containerd.Image) (string, error) {
+
+	// First make sure the dir exists on the host
+	configDir := path.Dir(configFile)
+
+	// TODO - consider refining this to be a generic utility to create a file on the host
+	// with the file content passed in as a string
+	testerSpec := specs.Spec{
+		Root: &specs.Root{
+			Path: "rootfs",
+		},
+		Process: &specs.Process{
+			Cwd: "/",
+			Args: []string{
+				"sh", "-c",
+				fmt.Sprintf("[ -f \"%s\" ] || echo \"{}\" > \"%s\" ; cat \"%s\"", configFile, configFile, configFile),
+			},
+			User: specs.User{
+				UID: 0,
+				GID: 0,
+			},
+			Env: []string{
+				"PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin",
+			},
+			NoNewPrivileges: false,
+		},
+	}
+	name := stringid.GenerateRandomID()[:20]
+	cOpts := []containerd.NewContainerOpts{
+		containerd.WithNewSnapshot(name, image),
+		containerd.WithImage(image),
+		containerd.WithSpec(&testerSpec,
+			containerized.WithHostMounts,
+			containerized.WithAllCapabilities,
+			containerized.WithHostPropagation,
+		),
+		containerized.WithHostDirMount(configDir, []string{"rbind", "rw", "rshared"}),
+	}
+	testerSpec.Mounts = []specs.Mount{
+		{
+			Options: []string{
+				"rbind",
+				"rw",
+				"rshared",
+			},
+			Source:      path.Dir(configFile),
+			Destination: path.Dir(configFile),
+			Type:        "bind",
+		},
+	}
+	container, err := c.cclient.NewContainer(
+		ctx,
+		name,
+		cOpts...,
+	)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to verify config file %s", configFile)
+	}
+	// Capture the output
+	buf := new(bytes.Buffer)
+	task, err := container.NewTask(ctx, defaultNewCreator(withByteBuffers(buf)))
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to verify %s", configFile)
+	}
+	err = task.Start(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to start container task")
+	}
+	status, err := task.Wait(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to wait for container task")
+	}
+	<-status
+	err = task.Kill(ctx, SIGKILL)
+	if err != nil {
+		logrus.Debugf("Failed to kill task: %s", err)
+	}
+	_, err = task.Delete(ctx)
+	if err != nil {
+		logrus.Debugf("Failed to delete task: %s", err)
+	}
+	err = container.Delete(ctx, containerd.WithSnapshotCleanup)
```

Comment:
Shouldn't the `task.Delete` and the `container.Delete` somehow defered? If an error occurs, the task and the container will remain?